test/e2e/daemonset/daemonset.go

```diff
@@ -0,0 +1,43 @@
+package daemonset
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type DaemonSetCommands struct{ daemonSet appsv1.DaemonSet }
+
+func (dc DaemonSetCommands) ReadyConditionEventually() AsyncAssertion {
+	return Eventually(func() bool {
+		return dc.daemonSet.Status.DesiredNumberScheduled == dc.daemonSet.Status.NumberAvailable
+	}, 60*time.Second, 1*time.Second)
+}
+
+func DaemonSet(daemonSetKey types.NamespacedName) DaemonSetCommands {
+	daemonSet := appsv1.DaemonSet{}
+	Eventually(func() appsv1.DaemonSet {
+		_ = framework.Global.Client.Get(context.TODO(), daemonSetKey, &daemonSet)
+		return daemonSet
+	}, 60*time.Second, 1*time.Second).ShouldNot(BeNil())
+	return DaemonSetCommands{daemonSet: daemonSet}
+}
+
+// GetDaemonSetList returns a DaemonSetList matching the labels passed
+func GetDaemonSetList(filteringLabels map[string]string) (appsv1.DaemonSetList, error) {
+	ds := appsv1.DaemonSetList{}
+	err := framework.Global.Client.List(context.TODO(), &ds, &client.ListOptions{LabelSelector: labels.SelectorFromSet(filteringLabels)})
+	return ds, err
+}
+
+// GetDaemonSet returns a DaemonSet matching the passed in DaemonSet name and namespace
+func GetDaemonSet(daemonSet appsv1.DaemonSet) (appsv1.DaemonSet, error) {
```

Comment:
pass as argument the key with not the daemonset with `types.NamespacedName`

Reply:
@bcrochet are you ok with this ?

Reply:
That's done.